### DISCUSSION TITLE
Install on Windows?

### DISCUSSION THREAD

**Discussion by SabinVI** [CyferShepard/Jellystat#224] [General]
Hello. Is there anyway to install Jellystat on Windows? I would really love to use this program, but unfortunately I don't run a Linux server. Also, I don't want to run Docker on Windows because I've read that there's a lot of wasted resources.

Every other Jellyfin related app can be directly installed in Windows. I am currently running:

- Jellyfin
- Jellyseer
- Sonarr
- Radarr
- Bazarr
- Prowlarr

Would love to add Jellystat to that list! :)

**Comment by SabinVI** [comment]
Bump... anyone?

I thought there was a way you could do it. I remember installing Jellyseer in Windows was able to be done even though there was no official installer. I believe I did it though the command prompt or Powershell using "yarn" commands. But it was a while ago so I don't really remember...

If anyone could help me with this I'd really appreciate it.

**Comment by SabinVI** [comment]
_Okay, so after my last post, I took a deep dive to see if I could figure out how to do it myself. I succeeded! Here's a guide to install Jellystat on Windows **without** Docker:_

# Installing Jellystat on Windows Server (Without Docker)

## Required Prerequisites

**1. Install Node.js**
* Download Node.js LTS version from `https://nodejs.org/`
* Run the installer with default options
* Verify installation by opening PowerShell and running:
```
node --version
npm --version
```
* Both commands should return version numbers

**2. Install PostgreSQL 16 from the official website**
* During installation, only "PostgreSQL Server" component required
* Remember your postgres password
* Keep default port (5432)
* Add PostgreSQL to your system PATH manually or restart your PC

**3. Install Git for Windows from the official website**
* Use default installation options

## Installation Steps

**1. Create PostgreSQL Database**

Open PowerShell as administrator and run:
```
psql -U postgres
```
When prompted for password, enter your PostgreSQL password.
Then create the database:
```
CREATE DATABASE jfstat;
```
Type `\q` to exit.

**2. Install Yarn**

In PowerShell as administrator:
```
npm install -g yarn
```

**3. Setup Jellystat**

In PowerShell as administrator:
```
cd C:\
mkdir jellystat
cd jellystat
git clone https://github.com/CyferShepard/Jellystat.git .
```

**4. Create Environment Files**

Copy this entire block and paste it into PowerShell (replace the postgres password with yours):
```
$envContent = @"
POSTGRES_USER=postgres
POSTGRES_PASSWORD="YourPostgresPasswordHere"
POSTGRES_IP=localhost
POSTGRES_PORT=5432
POSTGRES_DB=jfstat
JWT_SECRET=YOUR_SECRET_PHRASE_HERE_CAN_BE_ANYTHING
TZ=America/Los_Angeles
"@

$envContent | Out-File -FilePath "C:\jellystat\.env" -Encoding UTF8
$envContent | Out-File -FilePath "C:\jellystat\backend\.env" -Encoding UTF8
```

**5. Install Dependencies**

Still in the jellystat folder:
```
npm install
npm run build
```

**6. Choose ONE Startup Method:**

**Option A: Startup BAT File (Runs when user logs in)**

Copy this entire block and paste it into PowerShell:
```
$batContent = @"
@echo off
timeout /t 10
cd c:\jellystat
powershell -window hidden -command "npm run start-app"
"@

$startupPath = [System.IO.Path]::Combine([System.Environment]::GetFolderPath('Startup'), 'start-jellystat.bat')
$batContent | Out-File -FilePath $startupPath -Encoding ASCII
```

**Option B: Windows Service (Runs at system startup)**

In PowerShell as administrator:
```
# Install node-windows
npm install -g node-windows
cd C:\jellystat
npm link node-windows

# Create service script
$serviceContent = @"
const Service = require('node-windows').Service;
const svc = new Service({
    name: 'Jellystat',
    description: 'Jellystat Statistics App for Jellyfin',
    script: 'C:\\jellystat\\server\\index.js'
});

svc.on('install', function() {
    svc.start();
});

svc.install();
"@

$serviceContent | Out-File -FilePath "install-service.js" -Encoding UTF8

# Run the script
node install-service.js
```

To uninstall the service if needed:
```
# Create uninstall script
$uninstallScript = @"
const Service = require('node-windows').Service;
const svc = new Service({
    name: 'Jellystat',
    script: 'C:\\jellystat\\server\\index.js'
});
svc.on('uninstall',function(){console.log('Uninstalled.');});
svc.uninstall();
"@

$uninstallScript | Out-File -FilePath "uninstall-service.js" -Encoding UTF8
node uninstall-service.js
```

**7. Allow Through Firewall**

Run these commands in PowerShell:
```
New-NetFirewallRule -DisplayName "Jellystat-Frontend" -Direction Inbound -Action Allow -Protocol TCP -LocalPort 3000
New-NetFirewallRule -DisplayName "Jellystat-Backend" -Direction Inbound -Action Allow -Protocol TCP -LocalPort 3333
```

## Common Issues and Solutions

**"psql is not recognized as a command"**
* You need to add PostgreSQL to your system PATH or restart your PC

**"client password must be a string" error**
* Make sure your postgres password is in quotes in the .env files
* Verify .env exists in both the root jellystat folder AND the backend folder
* Double-check your postgres password is correct

**PostgreSQL not running**

Run in PowerShell:
```
Start-Service postgresql*
```

## Important Notes
* All PowerShell commands must be run as administrator
* If using Option A (bat file), the 10-second delay helps prevent resource conflicts
* If using Option B (service), it will start automatically with Windows
* This can run alongside Jellyseerr without issues
* Access Jellystat at `http://localhost:3000` after installation
* Keep a backup of your .env file
* For timezone, you can change America/Los_Angeles to your timezone
* The JWT_SECRET provided is secure - you don't need to change it

**This guide has been tested on Windows Server 2022. If you run into any issues, first verify that PostgreSQL is running and that all environment variables are set correctly in both .env files.**

**Reply by sipodshow** [reply]
i found this guide 
and followed it. but i have this issue:


changed 18 packages in 4s

2 packages are looking for funding
  run `npm fund` for details

up to date, audited 1878 packages in 15s

277 packages are looking for funding
  run `npm fund` for details

47 vulnerabilities (7 low, 20 moderate, 19 high, 1 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
node:fs:1349
  const result = binding.mkdir(
                         ^

Error: ENOENT: no such file or directory, mkdir 'C:\jellystat\server\daemon'
    at Object.mkdirSync (node:fs:1349:26)
    at C:\Users\zombi\AppData\Roaming\npm\node_modules\node-windows\lib\daemon.js:530:16
    at suppressedCallback (node:fs:254:5)
    at FSReqCallback.oncomplete (node:fs:185:23) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'mkdir',
  path: 'C:\\jellystat\\server\\daemon'
}

Node.js v24.11.1

im on a windows 10 pro desktop 
i am not an expert on any of the applications installed to achieve this so cant tell where it went wrong. is this supported here? please and thankyou

**Reply by SabinVI** [reply]
Hello,

Please try to paste any issues or error messages you get into ChatGPT, and it should be able to assist you.

As I was forming this guide, I used ChatGPT to guide me in areas I was not familiar with.

If you are able to get it to work, or need more help, please post here what happened so we can help others.

**Reply by SabinVI** [reply]
I actually pasted your issues into ChatGPT and Claude, and it may be that there was a bug in the code I listed for running JellyStat as a Windows service.

To be honest, I chose **Option A: Startup BAT File (Runs when user logs in)**

So, I was never able to test the part running as a Windows service. I know its not as pretty, but I have my Admin always auto logged in, so I just run it when the Admin user logs in.

If you need it run as a service, I'm sure ChatGPT or another AI can help you fix it. Please post the fix here if you can. I'd post it myself but it wasn't just a couple lines.

**Reply by zonkflut** [reply]
For the run as windows service, I changed
```
    script: 'C:\\jellystat\\server\\index.js' to script: 'C:\\jellystat\\backend\\server.js'
```

Full script:

```
# Install node-windows
npm install -g node-windows
cd C:\jellystat
npm link node-windows

# Create service script
$serviceContent = @"
const Service = require('node-windows').Service;
const svc = new Service({
    name: 'Jellystat',
    description: 'Jellystat Statistics App for Jellyfin',
    script: 'C:\\jellystat\\backend\\server.js'
});

svc.on('install', function() {
    svc.start();
});

svc.install();
"@

$serviceContent | Out-File -FilePath "install-service.js" -Encoding UTF8

# Run the script
node install-service.js
```
